lib/ansiblelint/utils.py

```diff
@@ -599,55 +599,109 @@ def get_first_cmd_arg(task):
 
 
 def append_skipped_rules(pyyaml_data, file_text, file_type):
-    """ Uses ruamel.yaml to parse comments then adds a
-        skipped_rules list to the task (or meta yaml block)
+    """Append 'skipped_rules' to individual tasks or single metadata block.
+
+    For a file, uses 2nd parser (ruamel.yaml) to pull comments out of
+    yaml subsets, check for '# noqa' skipped rules, and append any skips to the
+    original parser (pyyaml) data relied on by remainder of ansible-lint.
+
+    :param pyyaml_data: file text parsed via ansible and pyyaml.
+    :param file_text: raw file text.
+    :param file_type: type of file: tasks, handlers or meta.
+    :returns: original pyyaml_data altered with a 'skipped_rules' list added
+    to individual tasks, or added to the single metadata block.
     """
 
-    PLAYBOOK_TASK_KEYWORDS = [
-        'tasks',
-        'pre_tasks',
-        'post_tasks',
-        'handlers',
-    ]
+    try:
+        yaml_skip = _append_skipped_rules(pyyaml_data, file_text, file_type)
+    except Exception as exc:
+        # Notify user of skip error, do not stop, do not change exit code
+        print('Error trying to append skipped rules: {}'.format(repr(exc)))
+        return pyyaml_data
+    return yaml_skip
 
+
+def _append_skipped_rules(pyyaml_data, file_text, file_type):
+    # parse file text using 2nd parser library
     yaml = ruamel.yaml.YAML()
     ruamel_data = yaml.load(file_text)
 
+    # if file type is 'meta', add skips to metadata block and return
+    if file_type == 'meta':
+        skipped_rules = _get_rule_skips_from_yaml(ruamel_data)
```

Comment:
I believe that the conditional is not needed:
```suggestion
        pyyaml_data[0]['skipped_rules'] = _get_rule_skips_from_yaml(ruamel_data)
```

It also makes it easy for the further processing code to iterate over this thing without having to do extra checks for the dict key existence.